app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @posts = @user.posts.preload(:comments)
```

Comment:
- [ ] ActiveRecord provides an awesome method called `includes` which eager load association in a smarter way and prevents N+1 queries issues. We are still fetching the comments of each posts which still leads to the N+1 queries problems as you can see below. This problem can be resolve using includes methods: `@user.posts.includes(:comments)`

![Screenshot from 2023-10-03 04-49-27](https://github.com/Theodoraldo/blog/assets/63087491/6a132fda-7607-4092-bba4-729ec571d34f)